FirebaseAuth/Sources/Backend/RPC/MultiFactor/SignIn/FIRFinalizeMFASignInRequest.h

```diff
@@ -16,21 +16,22 @@
 
 #import "FirebaseAuth/Sources/Backend/FIRAuthRPCRequest.h"
 #import "FirebaseAuth/Sources/Backend/FIRIdentityToolkitRequest.h"
+#import "FirebaseAuth/Sources/Backend/RPC/Proto/FIRAuthProto.h"
 #import "FirebaseAuth/Sources/Backend/RPC/Proto/Phone/FIRAuthProtoFinalizeMFAPhoneRequestInfo.h"
+#import "FirebaseAuth/Sources/Backend/RPC/Proto/TOTP/FIRAuthProtoFinalizeMFATotpSignInRequestInfo.h"
 
 NS_ASSUME_NONNULL_BEGIN
 
 @interface FIRFinalizeMFASignInRequest : FIRIdentityToolkitRequest <FIRAuthRPCRequest>
 
 @property(nonatomic, copy, readonly, nullable) NSString *MFAPendingCredential;
 
-@property(nonatomic, copy, readonly, nullable)
-    FIRAuthProtoFinalizeMFAPhoneRequestInfo *verificationInfo;
+@property(nonatomic, copy, readonly, nullable) NSObject<FIRAuthProto> *verificationInfo;
 
-- (nullable instancetype)
-    initWithMFAPendingCredential:(NSString *)MFAPendingCredential
-                verificationInfo:(FIRAuthProtoFinalizeMFAPhoneRequestInfo *)verificationInfo
-            requestConfiguration:(FIRAuthRequestConfiguration *)requestConfiguration;
+- (nullable instancetype)initWithMFAPendingCredential:(NSString *)MFAPendingCredential
+                                     verificationInfo:(NSObject<FIRAuthProto> *)verificationInfo
+                                 requestConfiguration:
+                                     (FIRAuthRequestConfiguration *)requestConfiguration;
 
```

Comment:
I think there are some misunderstanding in that conversation.  Let's revisit them here.
My intention with updating the enrollment class from the LLD is to allow freedom for developer's work.  Since enrollment and sign-in does not impact each other, I think we can leave the code like this and here is why - 
- Allowing a base class for verification info would benefit any potential MFA factor we support in the future.  Means we do not have to update this class when every time there is a new feature coming up.
- Since the following implementation has already been done, it will be a huge amount work to refactor them.  Compare to the gain, the benefit is trivial (I assume the gain is to keep the consistency with enrollment class structure?).

Let me know if you disagrees.

Reply:
Thanks for looking into this. 
I agree with your first point but for the sake for code readability, it would be better to maintain consistency across code. The reason we preferred a separate function for enrollment instead of a parent class was because `FIRAuthProto` is a very broad class, so it could lead to more confusion compared to when these types are explicitly defined making it less error prone.

Reply:
First of all, let's be clear on what is readability and what is consistency.  Consistency means we are following a similar pattern when we design our work.  The most important part of consistency is to keep our external API align so our customers are having an easy experience.  Readability is a different concept that should focus on is your code easy to be understand, which normally is achieved by adding concise and clear comments and breaking down a long function.  In my opinion, using a base class adds more points to readability because in the method itself, developers can see the logic of this particular finailizeMFASignIn uses the same logic to call backend with different factors.

I also thought about create another base class just for MFA protos.  However, that class will be an empty class which leads to adding unnecessary code.  Hence prefered using FIRAuthProto.

I agree internal logic consistency also has its own benefit in code maintaining.  For our future implementing we can strictly follow our planning.

This is hard as I don't think the is a universal standard that fits all. As of now, my opinions stay with the benefit of the current implementations out rules the internal logic consistency. But if you feel strongly against it, feel free to update the enrollment flow to align with this change.  

As of error prone, let's make sure moving forward we have sufficient tests to avoid developing errors.

Also feel free to sync up with me offline so we can quickly move forward with this or file a bug to further discuss and address this.

Reply:
Thanks for shedding light on readability and consistency. I believe that both aspects are important and complement each other. While I still prefer strong types, I'm open to going ahead with the current format if you strongly believe it is the better approach. Let's ensure we have sufficient tests in place to minimize errors.